src/backend/services/sync/jobs/sync_agent.py

```diff
@@ -0,0 +1,42 @@
+from backend.config.tools import ToolName
+from backend.crud.agent import get_agent_by_id
+from backend.crud.agent_tool_metadata import get_all_agent_tool_metadata_by_agent_id
+from backend.database_models.database import get_session
+from backend.schemas.agent import Agent, AgentToolMetadata
+from backend.services.sync import app
+from backend.services.sync.constants import DEFAULT_TIME_OUT
+from backend.tools.google_drive import (
+    handle_google_drive_sync,
+    list_google_drive_artifacts_file_ids,
+)
+
+
+@app.task(time_limit=DEFAULT_TIME_OUT)
+def sync_agent(agent_id: str):
```

Comment:
lets say we've synced the files for the first time and now they're indexed into compass.

a user decides they don't want this file as part of their gdrive tool so they delete it and FE handles removing the agent metadata. do we want to also consider removing the index from compass (maybe we handle this case already but i missed it).

i can imagine a scenario where we a user syncs a ton of gdrive files, they get indexed into compass and then for whatever reason that agent is deleted, we'd be left with a ton of unused compass indices.

Reply:
In the roadmap